src/app/email/RenderEmail.tsx

```diff
@@ -0,0 +1,11 @@
+"use client";
+
+import { use } from "react";
+import { render } from "@react-email/render";
+
+export const RenderEmail = ({ children }: { children: React.ReactNode }) => {
+  const element = <>{children}</>;
+  const html = render(element);
+  const htmlString = use(html);
+  return <div dangerouslySetInnerHTML={{ __html: htmlString }} />;
+};
```

Comment:
We need this helper to ensure we are rendering using react-email's render instead of plain `react-dom/server` (when one uses `react` option for `resend.emails.send()`, it uses `@react-email/render`).

We just aren't able to do that directly, because we need to run this within the SSR context, otherwise we get RSC react and its a no-go.

Happy to elaborate :)